deepfence_worker/ingesters/common.go

```diff
@@ -36,69 +36,31 @@ func CommitFuncStatus[Status any](ts utils.Neo4jScanType) func(ns string, data [
-			MATCH (n) -[:SCANNED]- (r)
-			SET r.` + ingestersUtil.ScanCountField[ts] + `=count, r.` + ingestersUtil.ScanStatusField[ts] + `=n.status, r.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id`
-
-			others_query = `
-			UNWIND $batch as row
-			MATCH (n:` + string(ts) + `{node_id: row.scan_id})
-			SET n.status = row.scan_status, n.status_message = row.scan_message, n.updated_at = TIMESTAMP()
-			WITH n
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH n, count(m) as count
-			MATCH (n) -[:SCANNED]- (r)
-			SET r.` + ingestersUtil.ScanCountField[ts] + `=count, r.` + ingestersUtil.ScanStatusField[ts] + `=n.status, r.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id`
-
-		case utils.NEO4J_CLOUD_COMPLIANCE_SCAN:
-			in_progress_query = `
-			UNWIND $batch as row
-			MATCH (n:` + string(ts) + `{node_id: row.scan_id})
-			WHERE NOT n.status IN $cancel_states
-			SET n.status = row.scan_status, n.status_message = row.scan_message, n.updated_at = TIMESTAMP()
-			WITH n
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH n, count(m) as total_count
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH  n, total_count, m.resource as arn, count(m) as count
-			OPTIONAL MATCH (n) -[:SCANNED]- (cn) -[:OWNS]- (cr:CloudResource{arn: arn})
-			SET cn.` + ingestersUtil.ScanCountField[ts] + `=total_count, cn.` + ingestersUtil.ScanStatusField[ts] + `=n.status, cn.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id
-			SET cr.` + ingestersUtil.ScanCountField[ts] + `=count, cr.` + ingestersUtil.ScanStatusField[ts] + `=n.status, cr.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id`
-
-			others_query = `
-			UNWIND $batch as row
-			MATCH (n:` + string(ts) + `{node_id: row.scan_id})
-			SET n.status = row.scan_status, n.status_message = row.scan_message, n.updated_at = TIMESTAMP()
-			WITH n
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH n, count(m) as total_count
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH  n, total_count, m.resource as arn, count(m) as count
-			OPTIONAL MATCH (n) -[:SCANNED]- (cn) -[:OWNS]- (cr:CloudResource{arn: arn})
-			SET cn.` + ingestersUtil.ScanCountField[ts] + `=total_count, cn.` + ingestersUtil.ScanStatusField[ts] + `=n.status, cn.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id
-			SET cr.` + ingestersUtil.ScanCountField[ts] + `=count, cr.` + ingestersUtil.ScanStatusField[ts] + `=n.status, cr.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id`
-		}
+		query := `
+		UNWIND $batch as row
+		OPTIONAL MATCH (m) -[:DETECTED]- (n:` + string(ts) + `{node_id: row.scan_id}) -[:SCANNED]- (r)
+		WHERE NOT n.status IN $cancel_states
+		SET n.status = row.scan_status,
+			n.status_message = row.scan_message,
+			n.updated_at = TIMESTAMP(),
+			r.` + ingestersUtil.ScanStatusField[ts] + `=n.status,
+			r.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id,
+			r.` + ingestersUtil.ScanCountField[ts] + `=count(m)`
 
 		recordMap := statusesToMaps(data)
 		in_progress, others := splitInprogressStatus(recordMap)
```

Comment:
I was thinking about this earlier also. 
We can avoid the `splitInprogressStatus` function. We can do the split in the `statusesToMaps` function. 
From there, it can return 3 slices -> recordMap, in_progress and others.

Reply:
Both the pod update function & cloud resource update function need the full list, I think it's clearer and more re-usable that way